src/main/java/uk/gov/hmcts/reform/idam/web/AppController.java

```diff
@@ -480,8 +488,17 @@ public ModelAndView verification(@ModelAttribute("authorizeCommand") @Validated
 
         final boolean validationErrors = bindingResult.hasErrors();
         if (validationErrors) {
-            if (StringUtils.isEmpty(request.getCode())) {
+            final List<FieldError> codeErrors = ofNullable(bindingResult.getFieldErrors("code"))
+                .orElse(Collections.emptyList());
+            final List<String> errorCode = codeErrors.stream()
```

Comment:
I think a HashSet could be better here to make later contains calls quicker.

Reply:
correct: but it's a list with 3 items, shouldn't make a significant difference